docs/openrpc.json

```diff
@@ -0,0 +1,917 @@
+      },
+      "NanoContractAction": {
+        "type": "object",
+        "properties": {
+          "type": {
+            "enum": ["deposit", "withdrawal"]
+          },
+          "token": {
+            "$ref": "#/components/schemas/Token"
+          },
+          "amount": {
+            "$ref": "#/components/schemas/OutputValueType"
+          },
+          "address": {
+            "description": "When action type is 'withdrawal', address is required and specifies the destination for sending the output. When action type is 'deposit', address is optional and specifies the source from UTXOs.",
+            "$ref": "#/components/schemas/Address"
+          },
+          "changeAddress": {
+            "description": "When action type is 'deposit', change addess is optional and specifies the destination for the change output after selecting UTXOs. When action type is 'withdrawal', it is ignored.",
+            "$ref": "#/components/schemas/Address"
+          }
+        }
+      },
+      "TransactionHex": {
+        "description": "Transaction encoded in hexadecimal.",
+        "type": "string"
+      },
+      "NanoContract": {
+        "type": "object",
+        "allOf": [
+          {
+            "$ref": "#/components/schemas/Transaction"
+          },
+          {
+            "properties": {
+              "id": {
+                "type": "string"
+              },
+              "method": {
+                "type": "string"
+              },
+              "args": {
+                "type": "array",
+                "items": {
+                  "$ref": "#/components/schemas/Buffer"
+                }
+              },
+              "pubkey": {
+                "$ref": "#/components/schemas/Buffer"
+              },
+              "signature": {
+                "$ref": "#/components/schemas/Buffer"
+              }
+            }
+          }
+        ]
+      },
+      "Transaction": {
+        "type": "object",
+        "properties": {
+          "inputs": {
+            "type": "array",
+            "items": {
+              "$ref": "#/components/schemas/Input"
+            }
+          },
+          "outputs": {
+            "type": "array",
+            "items": {
+              "$ref": "#/components/schemas/Output"
+            }
+          },
+          "signalBits": {
+            "type": "number"
+          },
+          "version": {
+            "type": "number"
+          },
+          "weight": {
+            "type": "number"
+          },
+          "nonce": {
+            "type": "number"
+          },
+          "timestamp": {
+            "type": "number"
+          },
+          "parents": {
+            "type": "array",
+            "items": {
+              "type": "string"
+            }
+          },
+          "tokens": {
+            "type": "array",
+            "items": {
+              "$ref": "#/components/schemas/Token"
+            }
+          },
+          "hash": {
+            "type": "string"
```

Comment:
32 bytes hexadecimal. If we change the `Token` schema as in my suggestion, we can use it here and other places like this

Reply:
OK: change made:
- I created a new `TransactionId` schema.
- I replaced it within the `Token` schema, here and everywhere else where TransactionId appears.